### PR TITLE
fix: Replace NetcodePatcher action with MSBuild

### PR DESCRIPTION
**Pull request by Distractic**
## Checklist

- [x] The code is tested (if applicable)
- [x] The plugin works as expected in the game
- [x] The documentation is up to date (if needed)

## Description

According to the advice of [Lordfirespeed](https://github.com/Lordfirespeed) (the owner of the action used to patch the project), we need to use the library https://github.com/EvaisaDev/UnityNetcodePatcher. This library is maintained and allows us to patch locally the project.

❤️ Thank you!

### CODE REVIEW

File: Template/Template.csproj [hunk 1]
```diff
@@ -32,6 +32,20 @@
         <PackageReference Include="UnityEngine.Modules" Version="2022.3.9" IncludeAssets="compile"/>
     </ItemGroup>
 
+    <ItemGroup>
+        <!-- https://github.com/EvaisaDev/UnityNetcodePatcher -->
+        <!-- Allows to patch elements like networked behaviours, RPCs, etc. -->
+        <!-- Also, without that, the CI/CD will fail because the game's assemblies are not findable -->
+        <PackageReference Include="Evaisa.NetcodePatcher.MSBuild" Version="3.*" PrivateAssets="all"/>
+        <NetcodePatch Include="$(TargetPath)"/>
+    </ItemGroup>
```
Comment (Lordfirespeed) [inline Template/Template.csproj:41]: Remove lines 35-41 of this file - these are not necessary when using the CLI

Comment (Distractic) [reply Template/Template.csproj]: Removed

File: Template/Template.csproj [hunk 2]
```diff
@@ -32,16 +32,10 @@
...
-        <!-- Allows to patch elements like networked behaviours, RPCs, etc. -->
-        <!-- Also, without that, the CI/CD will fail because the game's assemblies are not findable -->
-        <PackageReference Include="Evaisa.NetcodePatcher.MSBuild" Version="3.*" PrivateAssets="all"/>
-        <NetcodePatch Include="$(TargetPath)"/>
-    </ItemGroup>
-
     <!-- https://github.com/EvaisaDev/UnityNetcodePatcher#usage-as-a-post-build-event -->
     <!-- Syntax to use the tool installed globally -->
+    <!-- Allows to patch elements like networked behaviours, RPCs, etc. -->
+    <!-- Also, without that, the CI/CD will fail because the game's assemblies are not findable -->
```
Comment (Lordfirespeed) [inline Template/Template.csproj:38]: This comment is misleading - it is actually [this line](https://github.com/Distractic/LethalCompanyTemplate/blob/cd2660489ffa13b844d70df9c0b203114b3f2eef/Common.props#L58) that is responsible for referencing the game's assemblies in CI environments

Comment (Distractic) [reply Template/Template.csproj]: Ah true, I mixed it up a bit

**Review by Lordfirespeed — APPROVED** [review]

**Comment by Lordfirespeed** [comment]
LGTM

**Pull request merged**